### ISSUE TITLE
Unable to reproduce results

### ISSUE THREAD

**Issue by pratik18v** [RuntianZ/macer#3]
Hi,
I was trying to reproduce the results reported in your paper. Specifically, I trained the following:

```
arch: ResNet-110
dataset: CIFAR10
sigma: 0.25
```

When performing certification using the entire test set, I get the ACR of **0.531**. When performing certification using the default arguments of the certify routine (`start_img=500, num_img=500, skip=1`) and using only 500 images in the test set, I get an ACR of **0.556**. This is exactly what is reported in the paper.

Can you please clarify if the ACR values reported for CIFAR-10 are generated for the whole test set or only 500 images?

Best,
Pratik

**Comment by pratik18v** [comment]
Hi,
In an effort to reproduce results from the paper, I used available pre-trained models (cifar10, sigma = 0.25) and performed certification using the code provided in this repository. The results are summarized below:

**MACER ResNet110**
ACR on 500 test images = 0.556 (identical to what's reported in the paper)
ACR on 10k test images = 0.531

**SmoothAdv ResNet110 (8-sample 10-step PGD with epsilon = 1.00)**
ACR on 500 test images = 0.565 (reported in paper as 0.538)
ACR on 10k test images = 0.551
Note: for this model, I had to change the normalization layer as advised by Salman et al.

In both cases (500 or 10k test images), SmoothAdv appears to be performing better than MACER. Can you please advise on how to obtain improvements over SmoothAdv using MACER?

Best,
Pratik
